### PR TITLE
OrthographicCamera Actor

### PR DESCRIPTION
**Pull request by MajorCooke**
This will make setting up orthographic cameras in map editors far easier. Once this is merged, I'll send a pull request over to Ultimate Doom Builder that implements support.

Arguments are as follows: 
- 0: Offset. This pushes the camera further away, going behind the camera. Default is 1.0 (converted to negative - the value cannot go lower than that).

### CODE REVIEW

File: wadsrc/static/zscript/actors/shared/camera.zs [hunk 1]
```diff
@@ -212,3 +212,35 @@ class SpectatorCamera : Actor
...
+	override void Tick()
+	{
+		if (current != args[0])
+			UpdateViewPos();
+		
+		Super.Tick();
+	}
+
+	private int current;
+	private void UpdateViewPos()
```
Comment (Boondorl) [inline wadsrc/static/zscript/actors/shared/camera.zs:241]: Don't use a private function in a virtual method. Those should be set to protected

Comment (MajorCooke) [reply wadsrc/static/zscript/actors/shared/camera.zs]: I see no reason to, but I went ahead and did as asked.

Comment (Boondorl) [reply wadsrc/static/zscript/actors/shared/camera.zs]: If they're private they cannot be modified from child class overrides. If the behavior shouldn't be modifiable it's better to bundle it in a protected function and call that from Tick() instead

**Comment by dileepvr** [comment]
Is this good @Boondorl ?

**Comment by Boondorl** [comment]
Seems fine to me

**Pull request merged**